src/language-js/printer-estree.js

```diff
@@ -3873,13 +3876,16 @@ function printMethodInternal(path, options, print) {
   return concat(parts);
 }
 
-function printJestEachTemplateLiteral(node, expressions, options) {
+function printJestEachTemplateLiteral(node, options, path, print) {
   /**
    * a    | b    | expected
    * ${1} | ${1} | ${2}
    * ${1} | ${2} | ${3}
    * ${2} | ${1} | ${3}
    */
+  options.isJestTemplateLiteral = true;
+  const expressions = path.map(print, "expressions");
+  options.isJestTemplateLiteral = false;
   const headerNames = node.quasis[0].value.raw.trim().split(/\s*\|\s*/);
   if (
     headerNames.length > 1 ||
```

Comment:
Please move printing the expressions inside this `if`. Print results are cached, so if this condition doesn't hold true , the other `print` call for the expressions (in `case "TemplateLiteral"`) will get the same results, which will be wrong.